### PR TITLE
ci: update CI image builder to use newer docker

### PR DESCRIPTION
**Pull request by crepererum**
This is a precondition to build ARM64 CI images.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -371,7 +371,8 @@ jobs:
   #
   # To modify the contents of the CI image, update docker/Dockerfile.ci
   ci_image:
-    machine: true
+    machine:
+      image: ubuntu-2004:202107-02
```
Comment (tustvold) [inline .circleci/config.yml:375]: Could you provide a link to where these are defined - it took me a lot of digging to not find a list :sweat_smile:

File: .circleci/config.yml [hunk 2]
```diff
@@ -371,7 +371,8 @@ jobs:
   #
   # To modify the contents of the CI image, update docker/Dockerfile.ci
   ci_image:
-    machine: true
+    machine:
```
Comment (tustvold) [inline .circleci/config.yml:374]: I'm not sure if there is a benefit to using a machine vs [setup_remote_docker](https://circleci.com/docs/2.0/building-docker-images/) as we do above. Perhaps there might be an advantage to being consistent?

**Review by tustvold — APPROVED** [review]
Left some comments you can take or leave

File: .circleci/config.yml (hunk 1 above)
Comment (crepererum) [reply .circleci/config.yml:380]: done

File: .circleci/config.yml (hunk 2 above)
Comment (crepererum) [reply .circleci/config.yml:378]: added a doc section explaining the reasoning

**Pull request merged**